File: java/src/test/java/dev/allwright/client/DialogTest.java
package dev.allwright.client;

import static org.junit.jupiter.api.Assertions.*;
import dev.allwright.engine.v1.DialogHookResult;
import dev.allwright.engine.v1.HookCompletedEvent;
import org.junit.jupiter.api.Test;

class DialogTest {
    @Test void decodesDialogMetadataAndRejectsInvalidResults() {
        var event = HookCompletedEvent.newBuilder().setDialog(DialogHookResult.newBuilder()
                .setDialogId("dialog").setType("prompt").setMessage("Name?").setDefaultValue("Ada")).build();
        var dialog = Hooks.DIALOG.decode(null, event);
        assertEquals("dialog", dialog.id());
        assertEquals("prompt", dialog.type());
        assertEquals("Name?", dialog.message());
        assertEquals("Ada", dialog.defaultValue());
        assertThrows(AllwrightException.class, () -> Hooks.DIALOG.decode(null, HookCompletedEvent.getDefaultInstance()));
    }
}

File: go/dialog_test.go
package allwright

import (
	enginev1 "allwright.dev/gen/allwright/engine/v1"
	"testing"
)

func TestDialogHookDecodesTypedResult(t *testing.T) {
	page := &Page{}
	result := &enginev1.HookCompletedEvent{Result: &enginev1.HookCompletedEvent_Dialog{Dialog: &enginev1.DialogHookResult{
		DialogId: "dialog", Type: "prompt", Message: "Name?", DefaultValue: "Ada",
	}}}
	dialog, err := Hooks.Dialog.decode(page, result)
	if err != nil {
		t.Fatal(err)
	}
	if dialog.Page() != page || dialog.ID() != "dialog" || dialog.Type() != "prompt" || dialog.Message() != "Name?" || dialog.DefaultValue() != "Ada" {
		t.Fatalf("unexpected dialog: %#v", dialog)
	}
	if _, err := Hooks.Dialog.decode(page, &enginev1.HookCompletedEvent{}); err == nil {
		t.Fatal("expected invalid result error")
	}
	if _, err := Hooks.Dialog.decode(&AndroidApp{}, result); err == nil {
		t.Fatal("expected unsupported owner error")
	}
}
